src/dialogs/mqtt-device-debug-info-dialog/dialog-mqtt-device-debug-info.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  LitElement,
+  html,
+  css,
+  CSSResult,
+  TemplateResult,
+  customElement,
+  property,
+} from "lit-element";
+import "@polymer/paper-dialog-scrollable/paper-dialog-scrollable";
+import "../../components/dialog/ha-paper-dialog";
+// Not duplicate, is for typing
+// tslint:disable-next-line
+import { HaPaperDialog } from "../../components/dialog/ha-paper-dialog";
+import "../../panels/config/mqtt/mqtt-device-debug-info-card";
+
+import { PolymerChangedEvent } from "../../polymer-types";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { MQTTDeviceDebugInfoDialogParams } from "./show-dialog-mqtt-device-debug-info";
+import { MQTTDeviceDebugInfo, fetchMQTTDebugInfo } from "../../data/mqtt";
+
+@customElement("dialog-mqtt-device-debug-info")
+class DialogMQTTDeviceDebugInfo extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() private _params?: MQTTDeviceDebugInfoDialogParams;
+  @property() private _debugInfo?: MQTTDeviceDebugInfo;
+  @property() private _error?: string;
+
+  public async showDialog(
+    params: MQTTDeviceDebugInfoDialogParams
+  ): Promise<void> {
+    this._params = params;
+    this._debugInfo = await fetchMQTTDebugInfo(this.hass, params.deviceId);
```

Comment:
Should this prevent the dialog from opening? This means that the user clicks "Show info" button and then has to wait until the data has been fetched, this can lead to the user clicking the button multiple times.

Reply:
OK, so how to do it instead?

Reply:
Call the fetch, but don't await it. Instead chain `.then(results => { this._debugInfo = results; })` to the fetchMQTTDebugInfo call.

Reply:
Fixed!